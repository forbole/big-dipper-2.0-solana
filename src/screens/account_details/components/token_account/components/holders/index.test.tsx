import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import Transactions from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Box: (props) => <div id="Box" {...props} />,
  TransactionsList: (props) => <div id="TransactionsList" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Transactions/List', () => {
  it('matches snapshot', async () => {
    let component;

    renderer.act(() => {
      component = renderer.create(
        <RecoilRoot>
          <MockTheme>
            <Transactions
              data={[
                {
                  address: 'address',
                  quantity: 5,
                  percentage: 30,
                  value: 24534,
                },
              ]}
              loadNextPage={jest.fn()}
              hasNextPage={false}
              isNextPageLoading={false}
            />
          </MockTheme>
        </RecoilRoot>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
