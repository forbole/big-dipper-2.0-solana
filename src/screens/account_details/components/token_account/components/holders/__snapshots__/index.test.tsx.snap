// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Transactions/List matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Box"
>
  <h2
    className="MuiTypography-root MuiTypography-h2"
  >
    accounts:holders
  </h2>
  <div
    className="makeStyles-list"
  >
    hello world
  </div>
</div>
`;
